src/Workspaces/Core/Portable/Workspace/Solution/Solution.CompilationTracker.cs

```diff
@@ -602,8 +590,16 @@ public CompilationInfo(Compilation compilation, bool hasCompleteReferences)
             {
                 try
                 {
-                    bool hasCompleteReferences = true;
+                    // if projectInfo.IsComplete is false, then this project is always not completed.
+                    bool isComplete = this.ProjectState.ProjectInfo.IsComplete;
+
                     var newReferences = new List<MetadataReference>();
+
+                    // we don't consider metadata reference being actually exist on disk as a part of complete reference
```

Comment:
I don't get this statement. ResolveAssemblyReferences has already filtered out references that don't exist on disk that are user-added. The only references we might find missing here are P2P ones where we most definitely might count that.